docs/understand/programming_model.rst

```diff
@@ -111,46 +139,215 @@ typically look the following:
+  :alt: Diagram depicting nested rectangles of varying color. The outermost one
+        titled "Grid", inside sets of uniform rectangles layered on one another
+        titled "Block". Each "Block" containing sets of uniform rectangles
+        layered on one another titled "Warp". Each of the "Warp" titled
+        rectangles filled with downward pointing arrows inside.
+
+  Hierarchy of thread groups.
+
+Warp (or Wavefront)
+  The innermost grouping of threads is called a warp, or a wavefront in ISA terms. A warp
+  is the most tightly coupled groups of threads, both physically and logically. Threads 
+  inside a warp are also called lanes, and the integral value identifying them is the lane ID. 
+  
+  .. tip::
+    Lane IDs aren't queried like other thread IDs, but are user-calculated. As a consequence they are only as multidimensional as the user interprets the calculated values to be.
+
+  The size of a warp is architecture dependent and always fixed. For AMD GPUs the wavefront is typically 64 threads, though sometimes 32 threads. Warps are signified by the set of communication primitives at their disposal, as discussed in :ref:`warp-cross-lane`.
+
+.. _inherent_thread_hierarchy_block:
+
+Block
+  The middle grouping is called a block or thread block. The defining feature
+  of a block is that all threads in a block will share an instance of memory
+  which they may use to share data or synchronize with one another.
+
+  The size of a block is user-configurable but is limited by the queryable
+  capabilities of the executing hardware. The unique ID of the thread within a
+  block is 3-dimensional as provided by the API. When linearizing thread IDs
+  within a block, assume the "fast index" being dimension ``x``, followed by
+  the ``y`` and ``z`` dimensions.
+
+.. _inherent_thread_hierarchy_grid:
+
+Grid
+  The outermost grouping is called a grid. A grid manifests as a single
+  dispatch of kernels for execution. The unique ID of each block within a grid
+  is 3-dimensional, as provided by the API and is queryable by every thread
+  within the block.
+
+Cooperative groups thread model
+-------------------------------
+
+The Cooperative groups API introduces new APIs to launch, group, subdivide,
+synchronize and identify threads, as well as some predefined group-collective
+algorithms, but most importantly a matching threading model to think in terms of.
+It relaxes some restrictions of the :ref:`inherent_thread_model` imposed by the
+strict 1:1 mapping of architectural details to the programming model. Cooperative
+groups let you define your own set of thread groups which may fit  your user-cases
+better than the defaults defined by the hardware.
+
+.. note::
+  The implicit groups defined by kernel launch parameters are still available 
+  when working with cooperative groups.
+
+For further information, see :doc:`Cooperative groups how-to <hip:how-to/cooperative_groups>`. 
```

Comment:
I understand the need to distinguish between the different categories of pages we have, but I think it's weird to have "how-to" just slapped at the end of sentence

Reply:
removed.